src/tool/xmeta/cpp/models/base_model.h

```diff
@@ -26,12 +31,65 @@ namespace xlang::xmeta
     struct method_model;
     struct event_model;
 
-    using class_type_semantics = std::variant<
+    struct formal_parameter_model;
+    struct symbol_table;
+    struct compilation_unit;
+    struct type_ref;
+
+    enum class compilation_error
+    {
+        passed = 0,
+        symbol_exists,
+        accessor_exists,
+        method_cannot_be_overloaded
+    };
+
+
+    enum class method_association
+    {
+        None,
+        Property,
+        Event,
+        Constructor
+    };
+
+    enum class simple_type
+    {
+        Boolean,
+        String,
+        Int8,
+        Int16,
+        Int32,
+        Int64,
+        UInt8,
+        UInt16,
+        UInt32,
+        UInt64,
+        Char16,
+        Single,
+        Double,
+    };
+
+    struct object_type {};
+
+    using class_type_semantics = std::variant<std::monostate,
+        std::shared_ptr<class_model>,
+        std::shared_ptr<enum_model>,
+        std::shared_ptr<interface_model>,
+        std::shared_ptr<struct_model>,
+        std::shared_ptr<delegate_model>,
+        std::shared_ptr<xlang::meta::reader::TypeDef>,
+        std::less<>>;
+
+    using type_semantics = std::variant<
         std::shared_ptr<class_model>,
         std::shared_ptr<enum_model>,
         std::shared_ptr<interface_model>,
         std::shared_ptr<struct_model>,
-        std::shared_ptr<delegate_model>>;
+        std::shared_ptr<delegate_model>,
+        std::shared_ptr<xlang::meta::reader::TypeDef>,
+        simple_type,
+        object_type>;
```

Comment:
I can't leave a comment on the "get_it" function below, but that's an awful function name. "get_by_id" would be better.

Reply:
Also, what does "id" refer to in this context. Is that a generalized concept for "name"?

Reply:
I have to agree. Will change the function name get_by_id. Id refers to the name of the member without the namespace part.

Reply:
maybe then "get_by_name", standardizing "name" to be "type name w/o namespace" and "qualified_name" to be "type name w/ namespace".  "fully_namespace_qualified_name" would be most descriptive, but that's too verbose I think .